github/status_test.go

```diff
@@ -15,57 +15,93 @@ import (
 	"github.com/Pix4D/cogito/help"
 )
 
-func TestGitHubStatusUseMockAPI(t *testing.T) {
+func TestGitHubStatusSuccessMockAPI(t *testing.T) {
 	cfg := help.FakeTestCfg
 	context := "cogito/test"
 	targetURL := "https://cogito.invalid/builds/job/42"
 	desc := time.Now().Format("15:04:05")
 	state := "success"
+
+	var testCases = []struct {
+		name   string
+		body   string
+		status int
+	}{
+		{
+			name:   "No errors",
+			body:   "Anything goes...",
+			status: http.StatusCreated,
+		},
+	}
+
+	for _, tc := range testCases {
+		ts := httptest.NewServer(
+			http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(tc.status)
+				fmt.Fprintln(w, tc.body)
+			}))
+		defer ts.Close()
+
+		t.Run(tc.name, func(t *testing.T) {
+			ghStatus := github.NewStatus(ts.URL, cfg.Token, cfg.Owner, cfg.Repo, context)
+			err := ghStatus.Add(cfg.SHA, state, targetURL, desc)
+			if err != nil {
+				t.Fatalf("\nhave: %s\nwant: <no error>", err)
+			}
+		})
+	}
+}
+
+func TestGitHubStatusFailureMockAPI(t *testing.T) {
+	cfg := help.FakeTestCfg
+	context := "cogito/test"
+	targetURL := "https://cogito.invalid/builds/job/42"
+	desc := time.Now().Format("15:04:05")
+	state := "success"
+
 	var testCases = []struct {
 		name       string
+		body       string
 		wantErr    string
 		wantStatus int
-		write      string
 	}{
 		{
-			"No errors",
-			"",
-			http.StatusCreated,
-			"Anything goes...",
+			name:       "Server error",
+			body:       "Something bad happened!",
+			wantErr:    http.StatusText(http.StatusInternalServerError),
+			wantStatus: http.StatusInternalServerError,
 		},
 		{
-			"Server error",
-			http.StatusText(500),
-			http.StatusInternalServerError,
-			"Something bad happened!",
-		},
-		{
-			"Repo not found",
-			fmt.Sprintf(`
+			name: "Repo not found",
```

Comment:
I'm wondering now if this is the correct name for this test. `not found` may hide insufficient rights. I don't have a good alternative here, may be `Not found or wrong token`?

Reply:
I see. Thinking about it, it seems that it is the "repo" name that is misleading. 
Taking a step back, the tests are exercising the GitHub Status API and are reacting to the HTTP status code received from the API. So the name could be simply "status: Not Found" or "interpret a 404 Not Found"

The other test, with `{"non existing repo: Not Found", `, is a bit different, because in that case the intent is really that the repo is non existing, as can be seen from the repo name ("non-existing-really"). On the other hand, it is true that probably when I wrote that test, we hadn't encountered yet this bizarre GitHub behaviour. For this other test, I will make a separate PR.

Reply:
Renamed as follows:

```
name: "500 Internal Server Error",
name: "404 Not Found (multiple causes)",
```